toolchain/diagnostics/diagnostic_emitter.h

```diff
@@ -302,6 +308,16 @@ struct DiagnosticTypeForArg<Arg> : public Arg::DiagnosticType {};
 
 }  // namespace Internal
 
+template <typename LocT>
+auto Emitter<LocT>::Builder::OverrideSnippet(llvm::StringRef snippet)
+    -> Builder& {
+  if (!emitter_) {
+    return *this;
+  }
+  diagnostic_.messages.back().loc.snippet = snippet;
```

Comment:
Shouldn't this have an error, and need to be:

```suggestion
  diagnostic_.messages.back().loc.snippet = snippet.str();
```

(Fine if I'm just wrong and the compiler is fine with this... wondering if the template is maybe not getting instantiated.)

Reply:
This works: looks like what's going on here is that `llvm::StringRef` implicitly converts to `std::string_view`, and `std::string` can be assigned from anything that implicitly converts to `std::string_view`.